cli/src/commands/extensions/api.rs

```diff
@@ -349,27 +353,34 @@ fn run_sandboxed(process: Process) -> Result<ProcessOutput> {
             };
 
             for path in process.exceptions.read.sandbox_paths().iter() {
-                permissions::add_exception(&mut birdcage, Exception::Read(PathBuf::from(path)))
+                let path = dirs::expand_home_path(path, &home_dir);
+                permissions::add_exception(&mut birdcage, Exception::Read(path))
                     .map_err(into_ioerr)?;
             }
             for path in process.exceptions.write.sandbox_paths().iter() {
-                permissions::add_exception(&mut birdcage, Exception::Write(PathBuf::from(path)))
+                let path = dirs::expand_home_path(path, &home_dir);
+                permissions::add_exception(&mut birdcage, Exception::Write(path))
                     .map_err(into_ioerr)?;
             }
             for path in process.exceptions.run.sandbox_paths().iter() {
-                let absolute_path = permissions::resolve_bin_path(path);
+                let absolute_path = permissions::resolve_bin_path(path).map_err(into_ioerr)?;
```

Comment:
I would have preferred resolving home dirs before bin_path, for consistency and to keep resolve_bin_path focused. I don't think we need this behavior anywhere else.

But I'm fine with both.

Reply:
Agreed, and reimplemented this way.